BTrees/tests/common.py

```diff
@@ -784,6 +876,24 @@ def testPop(self):
 class BTreeTests(MappingBase):
     # Tests common to all BTrees
 
+    def _getTargetClass(self):
+        # Most of the subclasses override _makeOne and not
+        # _getTargetClass, so we can get the type that way.
+        if type(self)._makeOne is not BTreeTests._makeOne:
+            return type(self._makeOne())
+
+        # Derive the class from the test case name, if not
+        # overridden
+        name = self.__class__.__name__
+        type_name = name[:-4]
```

Comment:
I'd like a short comment mentioning what suffix the `[:-4]` strips.
```python
type_name = name[:-4]  # e.g. OOBTreeTest -> OOBTree
```
Or, even better, an assertion
```python
assert name.endswith('Test')
type_name = name[:-len('Test')]
```